var/spack/repos/builtin/packages/wrf/package.py

```diff
@@ -15,132 +60,269 @@ class Wrf(Package):
-            env.set('FCFLAGS', args)
-            env.set('FFLAGS', args)
+        if self.spec.satisfies("%gcc@10:"):
+            args = "-w -O2 -fallow-argument-mismatch -fallow-invalid-boz"
+            env.set("FCFLAGS", args)
+            env.set("FFLAGS", args)
 
     def patch(self):
         # Let's not assume csh is intalled in bin
-        files = glob.glob('*.csh')
+        files = glob.glob("*.csh")
+
+        filter_file("^#!/bin/csh -f", "#!/usr/bin/env csh", *files)
+        filter_file("^#!/bin/csh", "#!/usr/bin/env csh", *files)
+
+    def answer_configure_question(self, outputbuf):
 
-        filter_file('^#!/bin/csh -f', '#!/usr/bin/env csh', *files)
-        filter_file('^#!/bin/csh', '#!/usr/bin/env csh', *files)
+        # Platform options question:
+        if "Please select from among the following" in outputbuf:
+            options = collect_platform_options(outputbuf)
+            comp_pair = "%s/%s" % (
+                basename(self.compiler.fc),
+                basename(self.compiler.cc),
+            )
+            compiler_matches = dict(
+                (x, y) for x, y in options.items() if comp_pair in x.lower()
+            )
+            if len(compiler_matches) > 1:
+                tty.warn("Found multiple potential build options")
+            try:
+                compiler_key = min(compiler_matches.keys(), key=len)
+                tty.warn("Selected build option %s." % compiler_key)
+                return (
+                    "%s\n"
+                    % compiler_matches[compiler_key][
+                        self.spec.variants["build_type"].value
+                    ]
+                )
+            except KeyError:
+                InstallError(
+                    "build_type %s unsupported for %s compilers"
+                    % (self.spec.variants["build_type"].value, comp_pair)
+                )
+
+        if "Compile for nesting?" in outputbuf:
+            options = collect_nesting_options(outputbuf)
+            try:
+                return "%s\n" % options[self.spec.variants["nesting"].value]
+            except KeyError:
+                InstallError("Failed to parse correct nesting option")
+
+    def do_configure_fixup(self):
+        # Fix mpi compiler wrapper aliases
+        if (
+            self.spec.satisfies("@3.9.1.1")
+            and self.spec.compiler.name == "gcc"
+        ):
+            rename(
+                "./arch/configure_new.defaults",
+                "./arch/configure_new.defaults.bak",
+            )
+            with open("./arch/configure_new.defaults.bak", "rt") as ifh, open(
+                "./arch/configure_new.defaults", "wt"
+            ) as ofh:
+                for line in ifh:
+                    if line.startswith("DM_"):
+                        line = line.replace("mpif90 -f90=$(SFC)", "mpif90")
+                        line = line.replace("mpicc -cc=$(SCC)", "mpicc")
```

Comment:
We could use `self.spec['mpi'].mpicc` and friends here instead, that may be more robust as not every MPI library uses the same names.

Reply:
Done